lib/hal/hal_impression_ac073tc1a.ex

```diff
@@ -0,0 +1,309 @@
+defmodule Inky.HAL.ImpressionAC073TC1A do
+  @default_io_mod Inky.IO.Impression
+
+  @moduledoc """
+  An `Inky.HAL` implementation responsible for sending commands to the Inky
+  screen. It delegates to whatever IO module its user provides at init, but
+  defaults to #{inspect(@default_io_mod)}
+  """
+
+  @behaviour Inky.HAL
+
+  @color_map %{black: 0, white: 1, green: 2, blue: 3, red: 4, yellow: 5, orange: 6, miss: 1}
+  # @colors %{
+  #   :black => 0,
+  #   :white => 1,
+  #   :green => 2,
+  #   :blue => 3,
+  #   :red => 4,
+  #   :yellow => 5,
+  #   :orange => 6,
+  #   # Not a color, but is used to clear the display
+  #   :clear => 7
+  # }
+
+  # PANEL SETTING
+  @psr 0x00
+  # POWER SETTING
+  @pwr 0x01
+  # POWER OFF
+  @pof 0x02
+  # POWER OFF SEQUENCE SETTING
+  @pofs 0x03
+  # POWER ON
+  @pon 0x04
+
+  # BTST1
+  @btst1 0x05
+  @btst2 0x06
+  # @dslp 0x07
+  @btst3 0x08
+
+  # DATA START TRANSMISSION 1
+  @dtm 0x10
+  # DISPLAY REFRESH
+  @drf 0x12
+
+  #?
+  @ipc 0x13
+
+  #?
+  @tse 0x41
+
+  # VCOM AND DATA INTERVAL SETTING
+  # This command indicates the interval of Vcom and data output. When setting
+  # the vertical back porch, the total blanking will be kept (20 Hsync).
+  @cdi 0x50
+  # TCON SETTING
+  # This command defines non-overlap period of Gate and Source.
+  @tcon 0x60
+  # RESOLUTION SETTING  (TRES)
+  # This command defines alternative resolution and this setting is of higher priority than the RES[1:0] in R00H (PSR).
+  @tres 0x61
+  # SPI FLASH CONTROL
+  # This command defines MCU host direct access external memory mode.
+  # This might allow us to specify our own lookup tables! Which might mean our own colors!
+  # @dam 0x65
+
+  # New in impression_7_3
+  @vdcs 0x82
+  @t_vdcs 0x84
+  @agid 0x86
+  @cmdh 0xAA
+  # @ccset 0xE0
+
+  @ccset 0xE0
+  @pws 0xE3
+  @tsset 0xE6
+
+  require Logger
+
+  alias Inky.Display
+  alias Inky.HAL
+  alias Inky.PixelUtil
+
+  defmodule State do
+    @moduledoc false
+
+    @state_fields [:display, :io_mod, :io_state, :setup?]
+
+    @enforce_keys @state_fields
+    defstruct @state_fields
+  end
+
+  #
+  # API
+  #
+
+  @impl HAL
+  def init(args) do
+    display = args[:display] || raise(ArgumentError, message: ":display missing in args")
```

Comment:
```suggestion
    display = args[:display] || raise(ArgumentError, message: "`:display` missing in args")
```

Back ticks here might improve clarity of this message.